content/en/blog/2024/new-otel-features-envoy-istio/index.md

```diff
@@ -0,0 +1,249 @@
+issue: 4534
+sig: OpenTelemetry Specification
+cSpell:ignore: bookinfo Grassi istioctl Joao productpage
+---
+
+In the dynamic world of cloud-native and distributed applications, managing
+microservices effectively is critical. [Kubernetes](https://kubernetes.io/) has
+become the de facto standard for container orchestration, enabling seamless
+deployment, scaling, and management of containerized applications.
+
+The distributed nature of such systems, however, adds a layer of complexity in
+the form of networking for in-cluster communication. Two well-known projects,
+Envoy and Istio, have emerged as the foundation for the smooth management and
+operation of such complex environments.
+
+Together, these technologies empower organizations to build scalable, resilient,
+and secure distributed systems.
+
+[Istio](https://istio.io/) is a service mesh, that orchestrates communication
+between microservices, providing features such as traffic management, security
+and, of course observability. Istio uses the Envoy proxy as its data plane.
+[Envoy](https://www.envoyproxy.io/) is a high-performance proxy, designed for
+single applications/services as well as a communication bus and "universal data
+plane" for service meshes.
+
+[Envoy](https://www.cncf.io/projects/envoy/) and
+[Istio](https://www.cncf.io/projects/istio/) projects are open source and part
+of the [Cloud Native Computing Foundation](https://www.cncf.io/).
+
+## Observability in Envoy and Istio
+
+The Envoy proxy deployed by the Istio service mesh is the perfect candidate to
+ensure incoming and outgoing requests are properly traced. This approach
+provides distributed traces of the entire service mesh, giving an overview on
+the communication between services — even when the applications themselves are
+not instrumented.
+
+> Note: At minimum, applications must be configured to propagate the
+> `traceparent` header.
+
+Envoy offers several
+[tracers](https://www.envoyproxy.io/docs/envoy/v1.29.4/api-v3/config/trace/trace)
+that do the job of tracing the requests, including the
+[OpenTelemetry tracer](/docs/concepts/signals/traces/#tracer). Tracers can be
+configured either directly within Envoy (when using it as a standalone
+component) or for all Envoy instances by using Istio.
+
+Here is an example of how Istio and Envoy work together to trace requests:
+
+![Distributed trace with Istio and Envoy](envoy-tracing.png)
+
+## New OTel tracing features in Envoy and Istio
+
+Although Envoy already had support for exporting OpenTelemetry traces using
+gRPC, it lacked support for exporting using HTTP. OpenTelemetry specifies both
+transports as first-class citizens. In addition, other areas such as providing
+resource attributes and configurable sampling decisions were lagging behind the
+stable portions of the OpenTelemetry specification.
+
+Starting from Envoy 1.29+ and Istio 1.21+, users have access to the new features
+described below.
```

Comment:
Because the sentence starts with "Starting" we don't need the "+" signs IMHO.

```suggestion
Starting from Envoy 1.29 and Istio 1.21, users have access to the new features
described below.
```

Reply:
Also, for interested readers who want to find out what is new in those releases, could you provide links to "Envoy 1.29" and "Istio 1.21"?

Reply:
Good idea! I added links to the changelog/release notes for each. Also spotted the wrong Istio version, it should be `1.22`. All fixed!